.github/actions/templates/publishModule/scripts/Publish-ModuleToPrivateBicepRegistry.ps1

```diff
@@ -0,0 +1,142 @@
+    }
+
+    process {
+        #############################
+        ##    EVALUATE RESOURCES   ##
+        #############################
+        if ((Split-Path $templateFilePath -Extension) -ne '.bicep') {
+            throw "The template in path [$templateFilePath] is no bicep template."
+        }
+
+        # Registry
+        if (-not (Get-AzContainerRegistry -ResourceGroupName $bicepRegistryRgName -Name $bicepRegistryName -ErrorAction 'SilentlyContinue')) {
+            if ($PSCmdlet.ShouldProcess("Container Registry [$bicepRegistryName] to resource group [$bicepRegistryRgName]", 'Deploy')) {
+                New-AzContainerRegistry -ResourceGroupName $bicepRegistryRgName -Name $bicepRegistryName -Sku 'Basic'
+            }
+        }
+
+        #################################
+        ##    FIND AVAILABLE VERSION   ##
+        #################################
+        $moduleRegistryIdentifier = 'bicep/modules/{0}' -f $moduleIdentifier.Replace('\', '/').Replace('/', '.').ToLower()
+        $repositories = Get-AzContainerRegistryRepository -RegistryName $bicepRegistryName
+        if ($repositories.Contains($moduleRegistryIdentifier)) {
+            $versions = (Get-AzContainerRegistryTag -RegistryName $bicepRegistryName -RepositoryName $moduleRegistryIdentifier).Tags.Name
+            $latestVersion = (($versions -as [Version[]]) | Measure-Object -Maximum).Maximum
+            Write-Verbose "Published versions detected in private bicep registry [$moduleIdentifier]. Fetched latest [$latestVersion]."
+        } else {
+            Write-Verbose "No version for module reference [$moduleRegistryIdentifier] detected in private bicep registry [$bicepRegistryName]. Creating new."
+            $latestVersion = New-Object System.Version('0.0.0')
+        }
+
+        ############################
+        ##    EVALUATE VERSION    ##
+        ############################
+
+        if (-not ([String]::IsNullOrEmpty($customVersion)) -and ((New-Object System.Version($customVersion)) -gt (New-Object System.Version($latestVersion)))) {
+            Write-Verbose "A custom version [$customVersion] was specified in the pipeline script and is higher than the current latest. Using it."
+            $newVersion = $customVersion
+        } else {
+            Write-Verbose 'No custom version set. Using default versioning.'
+
+            switch ($versioningOption) {
+                'major' {
+                    Write-Verbose 'Apply version update on "major" level'
+                    $newVersion = (New-Object -TypeName System.Version -ArgumentList ($latestVersion.Major + 1), $latestVersion.Minor, $latestVersion.Build).ToString()
+                    break
+                }
+                'minor' {
+                    Write-Verbose 'Apply version update on "minor" level'
+                    $newVersion = (New-Object -TypeName System.Version -ArgumentList $latestVersion.Major, ($latestVersion.Minor + 1), $latestVersion.Build).ToString()
```

Comment:
Patch version MUST be reset to 0 when minor version is incremented.

https://semver.org/#spec-item-7

```suggestion
                    $newVersion = (New-Object -TypeName System.Version -ArgumentList $latestVersion.Major, ($latestVersion.Minor + 1), 0).ToString()
```